### PR TITLE
Statistics query

### PR DESCRIPTION
**Pull request by fyellin**
Added implementation, docs, and tests for statistics queries.  I think this is the last remaining native wgpu feature not yet implemented.

### CODE REVIEW

File: docs/backends.rst [hunk 1]
```diff
@@ -44,8 +44,7 @@ It also works out of the box, because the wgpu-native DLL is shipped with wgpu-p
 
 The wgpu_native backend provides a few extra functionalities:
 
-.. py:function:: wgpu.backends.wgpu_native.request_device_sync(adapter, trace_path, *, label="", required_features, required_limits, default_queue)
-
+.. py:function:: wgpu.backends.wgpu_native.extras.request_device_sync(adapter, trace_path, *, label="", required_features, required_limits, default_queue)
```
Comment (almarklein) [inline docs/backends.rst:47]: These functions are defined in `extras` but can be imported from `from wgpu.backends.wgpu_native`.

File: tests/test_wgpu_statistics_query.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
+"""
+Test occlusion queries.
```
Comment (almarklein) [inline tests/test_wgpu_statistics_query.py:2]: Forgot to update title.

File: wgpu/enums.py [hunk 3]
```diff
@@ -409,3 +411,11 @@ class ErrorFilter(Enum):
...
     out_of_memory = "out-of-memory"
     internal = "internal"
+
+
+class PipelineStatisticName(Enum):  # wgpu native
+    VertexShaderInvocations = "vertex-shader-invocations"
+    ClipperInvocations = "clipper-invocations"
+    ClipperPrimitivesOut = "clipper-primitives-out"
+    FragmentShaderInvocations = "fragment-shader-invocations"
+    ComputeShaderInvocations = "compute-shader-invocations"
```
Comment (almarklein) [inline wgpu/enums.py:421]: We keep the main API free of backend-specific stuff like this. So we can do one of two things:

* Add an enum to `extras.py` (and import it in the `wgpu.backends.wgpu_native` namespace like the rest.
* Make it an "implicit enum" where we simply document for the `statistics` argument what the possible values are.

The changes to the code that generates `_mappings.py` is fine; we need to be able to convert these strings to ints.

**Review by almarklein — DISMISSED** [review]
Nice work! 🚀

**Review by Korijn — DISMISSED** [review]

**Comment by fyellin** [comment]
Someone with write access needs to merge this.

**Comment by almarklein** [comment]
I resolved the merge conflict.

**Review by almarklein — APPROVED** [review]

**Review by Korijn — APPROVED** [review]

**Pull request merged**